worker/provisioner/broker.go

```diff
@@ -100,7 +100,7 @@ func finishNetworkConfig(bridgeDevice string, interfaces []network.InterfaceInfo
 
 	if !haveNameservers || !haveSearchDomains {
 		logger.Warningf("incomplete DNS config found, discovering host's DNS config")
```

Comment:
this is the if that I think might be wrong. Do we always need both nameservers *and* search domains? Is it possible that we're overriding real, valid nameservers because we don't have a search domain configured?

Reply:
Hi @jameinel. Yes, that disjunction does leave things a bit uncertain but I dumped some debug info to see what was going on. The debug info revealed that, at least in the case of deploying to containers on MAAS 2.3, indeed no name servers were available. So in our case we indeed need to find nameservers here.